erpnext/stock/doctype/batch/batch.py

```diff
@@ -339,12 +339,57 @@ def validate_serial_no_with_batch(serial_nos, item_code):
 
 
 @frappe.whitelist()
-def save_thc_cbd(batch_no, thc, cbd):
-	doc = frappe.get_doc('Batch', batch_no)
-	doc.thc = thc
-	doc.cbd = cbd
-	doc.save()
-
+def update_batch_doc(batch_no, qi_name, item_code):
+
+	batch_doc = get_batch_fields(batch_no)
+	qi_doc = get_qi_fields(qi_name)
+	ci_doc = get_ci_fields(item_code)
+	readind_doc = get_readings_for_qi(qi_name)
+
+	frappe.db.set_value("Batch", batch_no, {
+		"thc" : qi_doc.thc,
+		"cbd" : qi_doc.cbd,
+		"sticker_details" : frappe.render_template(
+		"templates/includes/sticker_order_material_request.html", dict(
+			qi_doc = qi_doc,
+			ci_doc = ci_doc,
+			readind_doc = readind_doc,
+			batch_doc = batch_doc)
+		)
+	})
+
+def get_batch_fields(batch_no):
+	return frappe.db.get_value("Batch", batch_no, [
+		"manufacturing_date",
+		"harvest_date",
+		"packaged_date",
+	], as_dict=1)
+
+def get_qi_fields(qi_name):
+	return frappe.db.get_value("Quality Inspection",qi_name,[
+		"thc",
+		"cbd",
+		"testing_lab",
+		"testing_result_link",
+		"package_tag",
+		"testing_date",
+		"manufacturer_name",
+		"manufacturer_website",
+		"qty",
+		"uom",
+		"strain_notes",
+		"verified_by",
+	] , as_dict=1)
```

Comment:
At this point, you should just return the doc, lol

```suggestion
	return frappe.get_doc("Quality Inspection", qi_name).as_dict()
```